vscode/src/autoedits/debug-panel/debug-panel.ts

```diff
@@ -0,0 +1,163 @@
+ */
+export class AutoeditDebugPanel {
+    public static currentPanel: AutoeditDebugPanel | undefined
+    private static readonly viewType = 'codyAutoeditDebugPabel'
+
+    private readonly panel: vscode.WebviewPanel
+    private readonly extensionContext: vscode.ExtensionContext
+    private disposables: vscode.Disposable[] = []
+    private updatePending = false
+    private readonly throttleMs = 500 // Throttle updates to at most once per 500ms
+
+    private constructor(panel: vscode.WebviewPanel, extensionContext: vscode.ExtensionContext) {
+        this.panel = panel
+        this.extensionContext = extensionContext
+
+        // Set the webview's initial content
+        void this.updateContent()
+
+        // Listen for when the panel is disposed
+        // This happens when the user closes the panel or when the panel is closed programmatically
+        this.panel.onDidDispose(() => this.dispose(), null, this.disposables)
+
+        // Subscribe to store changes with throttling
+        this.disposables.push(
+            autoeditDebugStore.onDidChange(() => {
+                // If an update is already pending, don't schedule another one
+                if (!this.updatePending) {
+                    this.updatePending = true
+                    setTimeout(() => {
+                        this.updatePending = false
+                        void this.updateContent()
+                    }, this.throttleMs)
+                }
+            })
+        )
+
+        // Handle messages from the webview
+        this.panel.webview.onDidReceiveMessage(
+            message => {
+                if (message.type === 'ready') {
+                    // Send the initial data when the webview is ready
+                    void this.updateContent()
+                }
+            },
+            null,
+            this.disposables
+        )
+    }
+
+    /**
+     * Type-safe wrapper for sending messages to the webview.
+     * Ensures that only valid messages defined in the protocol are sent.
+     */
+    private postMessageToWebview(message: AutoeditDebugMessageFromExtension): void {
+        this.panel.webview.postMessage(message)
+    }
+
+    /**
+     * Shows the debug panel in the editor.
+     * If the panel already exists, it will be revealed.
+     */
+    public static showPanel(extensionContext: vscode.ExtensionContext): void {
+        const column = vscode.window.activeTextEditor
+            ? vscode.window.activeTextEditor.viewColumn
+            : undefined
+
+        // If we already have a panel, show it
+        if (AutoeditDebugPanel.currentPanel) {
+            AutoeditDebugPanel.currentPanel.panel.reveal(column)
+            return
+        }
+
+        // Create a new panel
+        const panel = vscode.window.createWebviewPanel(
+            AutoeditDebugPanel.viewType,
+            'Cody Auto-Edit Debug Panel',
+            column || vscode.ViewColumn.One,
```

Comment:
```suggestion
            vscode.ViewColumn.Beside,
```

Is this not preferred? Having to move it each time will get annoying. I'm not sure if there's a way to put in underneath, but this should always put it alongside the active editor I believe